openpype/hosts/houdini/plugins/publish/validate_review_colorspace.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+import pyblish.api
+from openpype.pipeline import (
+    PublishValidationError,
+    OptionalPyblishPluginMixin
+)
+from openpype.pipeline.publish import RepairAction
+from openpype.hosts.houdini.api.action import SelectROPAction
+from openpype.hosts.houdini.api.colorspace import get_default_display_view_colorspace  # noqa
+
+import os
+import hou
+
+
+class SetDefaultViewSpaceAction(RepairAction):
+    label = "Set default view colorspace"
+    icon = "mdi.monitor"
+
+
+class ValidateReviewColorspace(pyblish.api.InstancePlugin,
+                               OptionalPyblishPluginMixin):
+    """Validate Review Colorspace parameters.
+
+    It checks if 'OCIO Colorspace' parameter was set to valid value.
+    """
+
+    order = pyblish.api.ValidatorOrder + 0.1
+    families = ["review"]
+    hosts = ["houdini"]
+    label = "Validate Review Colorspace"
+    actions = [SetDefaultViewSpaceAction, SelectROPAction]
+
+    optional = True
+
+    def process(self, instance):
+
+        if not self.is_active(instance.data):
+            return
+
+        invalid = self.get_invalid(instance)
+        if invalid:
+            raise PublishValidationError(
+                ("'OCIO Colorspace' parameter is not valid."),
+                title=self.label
+            )
+
+    @classmethod
+    def get_invalid(cls, instance):
+
+        rop_node = hou.node(instance.data["instance_node"])
+        if os.getenv("OCIO") is None:
+            cls.log.debug(
+                "Default Houdini colorspace is used, "
+                " skipping check.."
+            )
+            return
+
+        if rop_node.evalParm("colorcorrect") != 2:
+            # any colorspace settings other than default requires
+            # 'Color Correct' parm to be set to 'OpenColorIO'
+            rop_node.setParms({"colorcorrect": 2})
```

Comment:
Noticed this just now - this should _not_ be in `get_invalid`  since it'd basically be changing nodes in your scene upon just validation. This should be in `repair()`